src/Symfony/Component/Mailer/Bridge/Sendgrid/Http/Api/SendgridTransport.php

```diff
@@ -67,7 +67,7 @@ private function getPayload(Email $email, SmtpEnvelope $envelope): array
         }
 
         $personalization = [
-            'to' => array_map($addressStringifier, $this->getRecipients($email, $envelope)),
+            'to' => array_map($addressStringifier, $email->getTo()),
```

Comment:
Are you sure we should not merge all emails from To/Cc/Bcc? And make them unique?

Reply:
Yes, the emails need to be unique between to, cc and bcc. Merging it all here means that the same email will appear in, for example, to and bcc (which errors on Sendgrid's API). Making it unique here means it needs to be removed from cc/bcc

Reply:
Does it mean that sendgrid reads the `Cc` and `Bcc` headers from the message directly? Then, why do we have to define `to` here? Can it be removed?

Reply:
> Does it mean that sendgrid reads the Cc and Bcc headers from the message directly? 

Yes, it reads it from the request object (not the headers)

> Then, why do we have to define to here? Can it be removed?

No, the Sendgrid api expects at least one `to` object, so it can't be removed. But it has dedicated fields for `cc` and `bcc`, so when including additional email addresses, those needs to stay in the `cc` or `bcc` fields and not be included in the `to` field.

Here is an example of what the api expects when including all the fields
https://sendgrid.com/docs/for-developers/sending-email/personalizations/#sending-a-single-email-to-a-single-recipient-with-a-cc-and-a-bcc

Reply:
So, would it be a good idea to also add the `cc` and `bcc` entries here?

Reply:
`cc` and `bcc` is already included below

```
if ($emails = array_map($addressStringifier, $email->getCc())) {
            $personalization['cc'] = $emails;
        }
        if ($emails = array_map($addressStringifier, $email->getBcc())) {
            $personalization['bcc'] = $emails;
        }
```

Reply:
Sorry for the noise :)